packages/plugin-coverage/src/lib/runner/lcov/merge-lcov.ts

```diff
@@ -0,0 +1,180 @@
+        ) {
+          return [...acc, [candidateRecord, candidateIndex]];
+        }
+        return acc;
+      },
+      [],
+    );
+
+    // This is not the first time the record has been identified as a duplicate
+    if (
+      duplicates.map(duplicate => duplicate[1]).some(index => index < currIndex)
+    ) {
+      return accMerged;
+    }
+
+    // Unique record
+    if (duplicates.length === 0) {
+      return [...accMerged, currRecord];
+    }
+
+    return [
+      ...accMerged,
+      mergeDuplicateLcovRecords([
+        currRecord,
+        ...duplicates.map(duplicate => duplicate[0]),
+      ]),
+    ];
+  }, []);
+}
+
+export function mergeDuplicateLcovRecords(records: LCOVRecord[]): LCOVRecord {
+  const linesDetails = mergeLcovLineDetails(
+    records.map(record => record.lines.details),
+  );
+  const linesHit = linesDetails.reduce(
+    (acc, line) => acc + (line.hit > 0 ? 1 : 0),
+    0,
+  );
+
+  const branchesDetails = mergeLcovBranchesDetails(
+    records.map(record => record.branches.details),
+  );
+  const branchesHit = branchesDetails.reduce(
+    (acc, branch) => acc + (branch.taken > 0 ? 1 : 0),
+    0,
+  );
+
+  const functionsDetails = mergeLcovFunctionsDetails(
+    records.map(record => record.functions.details),
+  );
+
+  const functionsHit = functionsDetails.reduce(
+    (acc, func) => acc + (func.hit != null && func.hit > 0 ? 1 : 0),
+    0,
+  );
+
+  const mergedRecord: LCOVRecord = {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    file: records[0]!.file,
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    title: records[0]!.title,
+    lines: {
+      found: linesDetails.length,
+      hit: linesHit,
+      details: linesDetails,
+    },
+    branches: {
+      found: branchesDetails.length,
+      hit: branchesHit,
+      details: branchesDetails,
+    },
+    functions: {
+      found: functionsDetails.length,
+      hit: functionsHit,
+      details: functionsDetails,
+    },
+  };
+  return mergedRecord;
+}
+
+export function mergeLcovLineDetails(
+  details: LinesDetails[][],
+): LinesDetails[] {
+  const flatDetails = details.flat();
+
+  const uniqueLines = [
+    ...new Set(flatDetails.map(flatDetail => flatDetail.line)),
+  ];
+
+  return uniqueLines.map(line => {
+    const hitSum = flatDetails
+      .filter(lineDetail => lineDetail.line === line)
+      .reduce((acc, lineDetail) => acc + lineDetail.hit, 0);
+
+    return { line, hit: hitSum };
+  });
+}
+
+export function mergeLcovBranchesDetails(
+  details: BranchesDetails[][],
+): BranchesDetails[] {
+  const flatDetails = details.flat();
+
+  const uniqueBranches = [
+    ...new Set(
+      flatDetails.map(({ line, block, branch }) =>
+        JSON.stringify({ line, block, branch }),
+      ),
+    ),
+  ].map(
+    functionJSON =>
+      JSON.parse(functionJSON) as {
+        line: number;
+        block: number;
+        branch: number;
+      },
```

Comment:
This way the object type is derived from the original source, which makes it a bit safer:

```suggestion
      JSON.parse(functionJSON) as Pick<BranchesDetails, 'line' | 'block' | 'branch'>,
```

Reply:
Good point!